ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -105,24 +110,18 @@ static int component_register(void) {
 }
 
 static int progress_callback(void) {
-    if (mca_osc_ucx_component.ucp_worker != NULL &&
-        mca_osc_ucx_component.num_incomplete_req_ops > 0) {
+    if ((mca_osc_ucx_component.ucp_worker != NULL) &&
```

Comment:
maybe we could register/unregister this progress callback on-demand? to avoid the overhead of function call

Reply:
could be issue if we unregister callback during call from callback, not sure how ompi will fail

Reply:
could we destroy the module during call from callback?

Reply:
yeah, seems like _opal_progress_register() is safe to call from the callback, for example ob1 is doing that (in "add_request_to_send_pending" pml_ob1_sendreq.h)